### PR TITLE
[dhd] Helpers cleanup

### PR DESCRIPTION
**Pull request by Thaodan**
- Use builtin $PWD
- Fix quoting in case path names contain spaces [SC2046]
- Remove useless function keyword 
- Remove useless true just return zero
- Use if directly, using test is redundant
- Remove yesnoal, add --build-all
- Quote ${BUILDSPEC_FILE[@]} to avoid resplitting of elements
- Don't use bash == on posix test [] and use [] everywhere

### CODE REVIEW

**Comment by Thaodan** [comment]
rebased against master

File: helpers/build_packages.sh [hunk 1]
```diff
@@ -83,6 +83,9 @@ while true; do
               *) BUILDPKG_PATH=$2;;
           esac
           shift;;
+      --build-all)
+          BUILDALL=1
```
Comment (sledges) [inline helpers/build_packages.sh]: When `--build-all` is not specified in command line, the script still builds all packages.

Also please mention this new flag in the `usage()`.

**Comment by Thaodan** [comment]
Rebased and changed as requested.

File: helpers/build_packages.sh (hunk 1 above)
Comment (sledges) [reply helpers/build_packages.sh]: Thanks for rebase, I took those changes and ran the script without --build-all again, and it still builds all packages:)

**Comment by mlehtima** [comment]
Nothing uses the BUILDALL variable in the code after you removed yesnoall, I doubt that is intended? Also it might not be a good idea to completely remove the possibility to skip building certain packages as often porters have to skip packages because they have to build manually (build_packages.sh -b) an older version of some packages due to incompatibility with current Sailfish OS version. Maybe changing yesnoall to yesno or just keeping it would make more sense.

**Comment by Thaodan** [comment]
It was because I've missed add another fix after that.
I refaktored build all  to just build every package (mw, hal and so on).
My original ideas to remove the questions every time a package is build 
with -b and keep the old behaivior when --build-all is used.

I for my self prefer to use --build every time to build pkgs in the order I want 
and to have a better overview in my build system as shown here:
https://gitlab.com/Thaodan/hadk_tools/blob/kumano/templates/
mer.droidmedia.hadk[1] 

--------
[1] https://gitlab.com/Thaodan/hadk_tools/blob/kumano/templates/
mer.droidmedia.hadk

**Comment by sledges** [comment]
This has become too convoluted. BTW your last version of the yesnoall-removing commit has reverted all previous PR's commits.

And now that I understand what you originally meant, the "--build-all" flag, it should've be called "--build-all-mw", because "building of all" packages (dhd+cfg+mw+ver) is already there by default when not providing any cmdline arguments.

Please remove the "yesnoall" commit from this PR (because it is called 'Helpers cleanup') so the cleanup commits could be merged, and please create a separate PR for such functionality change.

The more useful behaviour to reduce the usage of yesnoall IMHO could be:
- `build_packages.sh` (no arguments) = builds all packages and doesn't ask yesnoall during mw (=unassisted build)
- `build_packages.sh --mw` = builds mw packages and asks for yesnoall
- `build_packages.sh --build` = doesn't ask the superfluous singleton yesnoall question
This means no changes to the actual `usage()`.

Comments to this idea welcome.

**Comment by Thaodan** [comment]
I like the idea but why ask during --mw builds?

**Comment by sledges** [comment]
Found this in the commit body 3a26d3422188e82d7d2ccef7e08543d7f2c2b096 :grinning:

**Comment by Thaodan** [comment]
Fixed.

**Comment by sledges** [comment]
Here's my take on the build all solution:

```diff
diff --git a/helpers/build_packages.sh b/helpers/build_packages.sh
index c47edfb..758aacb 100755
--- a/helpers/build_packages.sh
+++ b/helpers/build_packages.sh
@@ -62,6 +62,7 @@ fi
 eval set -- $OPTIONS
 
 if [ "$#" == "1" ]; then
+    BUILDALL=1
     BUILDDHD=1
     BUILDCONFIGS=1
     BUILDMW=1
@@ -117,10 +118,10 @@ if [ "$BUILDCONFIGS" = "1" ]; then
         sb2 -t $VENDOR-$DEVICE-$PORT_ARCH -m sdk-install -R zypper se -i community-adaptation > /dev/null
         ret=$?
         if [ $ret -eq 104 ]; then
-            BUILDALL=y
+            BUILDALL_OVERRIDE=1
             buildmw -u "https://github.com/mer-hybris/community-adaptation.git" \
                     -s rpm/community-adaptation-localbuild.spec || die
-            BUILDALL=n
+            BUILDALL_OVERRIDE=0
         elif [ $ret -ne 0 ]; then
             die "Could not determine if community-adaptation package is available, exiting."
         fi
diff --git a/helpers/util.sh b/helpers/util.sh
index 97ce2fd..3318af8 100755
--- a/helpers/util.sh
+++ b/helpers/util.sh
@@ -33,7 +33,7 @@
 #
 
 PORT_ARCH="${PORT_ARCH:-armv7hl}"
-BUILDALL=n
+BUILDALL=0
 LOG="/dev/null"
 CREATEREPO="createrepo_c"
 ALLOW_UNSIGNED_RPM=""
@@ -139,7 +139,7 @@ buildversion() {
 }
 
 yesnoall() {
-    if [ $BUILDALL = "y" ]; then
+    if [ $BUILDALL = "1" ] || [ $BUILDALL_OVERRIDE = "1" ]; then
         return 0
     fi
     read -r -p "${1:-} [Y/n/all]" REPLY
@@ -149,7 +149,7 @@ yesnoall() {
        true
        ;;
     [aA]*)
-       BUILDALL=y
+       BUILDALL=1
        true
        ;;
     *)
```

**Comment by Thaodan** [comment]
Mine is at #236. It works the similar but only when the option is set to ask during the build of a pkg. 
Should this apply only to --mw builds or to other section to such as --hal?

**Comment by sledges** [comment]
Tested, works, thanks.

**Comment by Thaodan** [comment]
Changed as requested

**Comment by sledges** [comment]
> Should this apply only to --mw builds or to other section to such as --hal?

It always applied to --mw builds only https://github.com/mer-hybris/droid-hal-device/blob/8a0564f1a74cdc71c81725a79a72d745accfe813/helpers/util.sh#L190

Such functionality should not change. It's only the mw packages that the user cannot select which ones to build from cmdline. They can always combine `-d, -c, and -v` switches as needed, but not mw.

**Pull request merged**